### PR TITLE
upgrade: Download the latest crowbar_join from the server

### PR DESCRIPTION
**Pull request by jsuchome**
Download and execute latest crowbar_join from the provisioner server
so it is not rewritten during chef-client run.

### CODE REVIEW

File: chef/cookbooks/crowbar/templates/default/crowbar-chef-upgraded.sh.erb [hunk 1]
```diff
@@ -26,8 +26,11 @@ if [[ -f $UPGRADEDIR/crowbar-chef-upgraded-ok ]] ; then
     exit 0
 fi
 
+# Download the latest crowbar-join from the server
+curl -s -o /usr/sbin/crowbar_join_tmp <%= @crowbar_join %>
```
Comment (vuntz) [inline chef/cookbooks/crowbar/templates/default/crowbar-chef-upgraded.sh.erb]: You also need to chmod a+x the file.

**Comment by jsuchome** [comment]
[Fixing upgrade breakage caused by rewritten crowbar_join](https://trello.com/c/MbTQpr56/227-fixing-upgrade-breakage-caused-by-rewritten-crowbar-join)

File: chef/cookbooks/crowbar/templates/default/crowbar-chef-upgraded.sh.erb (hunk 1 above)
Comment (jsuchome) [inline chef/cookbooks/crowbar/templates/default/crowbar-chef-upgraded.sh.erb]: Added `+x` (should be enough)

**Comment by rsalevsky** [comment]
Could you add the bsc number?

**Comment by jsuchome** [comment]
Tested manually, seems to solve the problem

**Comment by jsuchome** [comment]
Normal gating job is somehow stuck, but notice that the upgrade one has passed

**Pull request merged**